ui/app/components/auth/form/base.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+
+import type AuthService from 'vault/vault/services/auth';
+import type ClusterModel from 'vault/models/cluster';
+import type { HTMLElementEvent } from 'vault/forms';
+
+/**
+ * @module Auth::Base
+ *
+ * @param {string} authType - chosen login method type
+ * @param {object} cluster - The cluster model which contains information such as cluster id, name and boolean for if the cluster is in standby
+ * @param {function} onError - callback if there is a login error
+ * @param {function} onSuccess - calls onAuthResponse in auth/page redirects if successful
+ */
+
+interface Args {
+  authType: string;
+  cluster: ClusterModel;
+  onError: CallableFunction;
+  onSuccess: CallableFunction;
+}
+
+export default class AuthBase extends Component<Args> {
+  @service declare readonly auth: AuthService;
+
+  @action
+  onSubmit(event: HTMLElementEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const formData = new FormData(event.target as HTMLFormElement);
+    const data: Record<string, FormDataEntryValue | null> = {};
+
+    for (const key of formData.keys()) {
+      data[key] = formData.get(key);
+    }
+
+    this.login.unlinked().perform(data);
+  }
+
+  login = task(
+    waitFor(async (data) => {
+      try {
+        const authResponse = await this.auth.authenticate({
+          clusterId: this.args.cluster.id,
+          backend: this.args.authType,
+          data,
+          selectedAuth: this.args.authType,
+        });
+
+        // responsible for redirect after auth data is persisted
+        this.onSuccess(authResponse);
+      } catch (error) {
+        this.onError(error as Error);
+      }
+    })
+  );
+
+  // if we move auth service authSuccess method here (or to each auth method component)
+  // then call that before calling parent this.args.onSuccess
+  onSuccess(authResponse: object) {
+    //  responsible for redirect after auth data is persisted
+    this.args.onSuccess(authResponse, this.args.authType);
+  }
+
+  onError(error: Error) {
+    if (!this.auth.mfaErrors) {
```

Comment:
I made this plural instead of singular like it is in `login-form.js` https://github.com/hashicorp/vault/blob/741721ad0e2f1888114941a3009133cccdc4b7b2/ui/app/components/auth/login-form.js#L91-L93

I think this may have been a typo [in the original PR](https://github.com/hashicorp/vault/pull/14145/files#diff-5c9db73e13f1501c4a5505b12a8a7bc143891a09e6e39bb111f3107edc2b79e7R217-R219), the auth service only has `mfaErrors` plural, there's no singluar `mfaError` so I updated it here, but have a note locally to come back to and test. I think it hasn't been an issue because a different component renders for MFA so it's unlikely to hit this conditional.